qcodes/instrument/base.py

```diff
@@ -114,24 +118,30 @@ def add_parameter(self, name: str,
                 unit of the new parameter is inconsistent with the existing
                 one.
         """
-        existing_parameter = self.parameters.get(name, None)
-
-        if existing_parameter:
+        if "bind_to_instrument" not in kwargs.keys():
+            kwargs["bind_to_instrument"] = True
 
-            if not existing_parameter.abstract:
-                raise KeyError(f"Duplicate parameter name {name}")
-
-            existing_unit = getattr(existing_parameter, "unit", None)
-            new_unit = kwargs.get("unit", None)
-            if existing_unit != new_unit:
-                raise ValueError(
-                    f"The unit of the parameter '{name}' is '{new_unit}'. "
-                    f"This is inconsistent with the unit defined in the "
-                    f"base class"
-                )
+        try:
+            param = parameter_class(name=name, instrument=self, **kwargs)
+        except TypeError:
+            kwargs.pop("bind_to_instrument")
+            warnings.warn(
+                f"Parameter {name} on instrument {self.name} does "
+                f"not correctly pass kwargs to its baseclass. A "
+                f"Parameter class must take `**kwargs` and forward "
+                f"them to its baseclass."
+            )
+            param = parameter_class(name=name, instrument=self, **kwargs)
```

Comment:
Need a test for this corner case. And need to verify that all parameter subclasses in qcodes allows forwarding of kwargs,